GopherCAN_ring_buffer.h

```diff
@@ -22,12 +22,14 @@ typedef struct
 
 // function prototypes
 void init_buffer(CAN_MSG_RING_BUFFER* buffer, CAN_MSG buffer_memory_ptr[], U8 buffer_size);
-boolean is_full(CAN_MSG_RING_BUFFER* buffer);
-boolean is_empty(CAN_MSG_RING_BUFFER* buffer);
-CAN_MSG* get_from_buffer(CAN_MSG_RING_BUFFER* buffer, U8 index);
 void remove_from_front(CAN_MSG_RING_BUFFER* buffer);
 void add_message_to_back(CAN_MSG_RING_BUFFER* buffer, CAN_MSG* message);
+void add_message_by_highest_prio(CAN_MSG_RING_BUFFER* buffer, CAN_MSG* message);
 
+// some handy macro functions
+#define IS_FULL(buffer) ((buffer)->fill_level >= (buffer)->size)
```

Comment:
These make everything a lot neater, but be careful here. You need to ensure you don't change the fields of the struct or this will break in new and exciting ways